Makefile

```diff
@@ -63,6 +63,10 @@ ifeq ($(CONFIG_BUILD_KERNEL),y)
 install: $(foreach SDIR, $(CONFIGURED_APPS), $(SDIR)_install)
 
 .import: $(foreach SDIR, $(CONFIGURED_APPS), $(SDIR)_all)
+	$(Q) for app in ${CONFIGURED_APPS}; do \
+		$(MAKE) -C "$${app}" archive ; \
```

Comment:
archive isn't a general target, why we need this?

Reply:
@xiaoxiang781216 
We need the archive target to generate libapps.a which is used for the init (the nsh)

Reply:
But, I can't find archive target from the top Makefile or Application.mk.

Reply:
> But, I can't find archive target from the top Makefile or Application.mk.

@xiaoxiang781216 
You can find the target at
https://github.com/apache/incubator-nuttx-apps/blob/3c556384ea685ebb12d3c6cb7f1c34c07b4d2015/Application.mk#L144